### PR TITLE
[CosmosDb] Scaler APIs

### PR DESCRIPTION
**Pull request by alrod**

### CODE REVIEW

File: src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDbScalerProvider.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            public string Connection { get; set; }
+
+            [JsonProperty]
+            public string DatabaseName { get; set; }
+
+            [JsonProperty]
+            public string ContainerName { get; set; }
+
+            [JsonProperty]
+            public string LeaseContainerName { get; set; }
```
Comment (ealsur) [inline src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDbScalerProvider.cs:68]: You also need the LeaseDatabaseName, users can customize this too. If not present, then the DatabaseName is used

File: src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDbScalerProvider.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            [JsonProperty]
+            public int MaxItemsPerInvocation { get; set; }
+
+            public void ResolveProperties(INameResolver resolver)
+            {
+                if (resolver != null)
+                {
+                    DatabaseName = resolver.ResolveWholeString(DatabaseName);
+                    ContainerName = resolver.ResolveWholeString(ContainerName);
+                    LeaseContainerName = resolver.ResolveWholeString(LeaseContainerName);
```
Comment (ealsur) [inline src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDbScalerProvider.cs:85]: LeaseContainerName is optional. If the user is not setting it, we use "leases" as default value -> https://github.com/Azure/azure-webjobs-sdk-extensions/blob/dev/src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerConstants.cs#L8

File: test/WebJobs.Extensions.CosmosDB.Tests/WebJobs.Extensions.CosmosDB.Tests.csproj [hunk 3]
```diff
@@ -8,7 +8,11 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.Storage" Version="4.0.3" />
+    <PackageReference Include="Microsoft.Azure.WebJobs" Version="3.0.37" />
+    <PackageReference Include="Microsoft.Azure.WebJobs.Host.Storage" Version="5.0.0" />	  
+    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.Storage" Version="4.0.3" />	  
+    <PackageReference Include="Microsoft.Azure.WebJobs.Host.TestCommon" Version="3.0.30" />	  
+    <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="7.0.0" />
```
Comment (ealsur) [inline test/WebJobs.Extensions.CosmosDB.Tests/WebJobs.Extensions.CosmosDB.Tests.csproj]: nit: Is version 7 needed? Shouldn't it be 6 (LTS)?

File: src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDbScalerProvider.cs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            }
+
+            ILoggerFactory loggerFactory = serviceProvider.GetService<ILoggerFactory>();
+            CosmosDbMetadata cosmosDbMetadata = JsonConvert.DeserializeObject<CosmosDbMetadata>(triggerMetadata.Metadata.ToString());
+            cosmosDbMetadata.ResolveProperties(serviceProvider.GetService<INameResolver>());
+            IOptions<CosmosClientOptions> options = serviceProvider.GetService<IOptions<CosmosClientOptions>>();
+            ICosmosDBServiceFactory serviceFactory = serviceProvider.GetService<ICosmosDBServiceFactory>();
+            CosmosClient cosmosClient = serviceFactory.CreateService(cosmosDbMetadata.Connection, options.Value);
+            var monitoredContainer = cosmosClient.GetContainer(cosmosDbMetadata.DatabaseName, cosmosDbMetadata.ContainerName);
+            var leaseContainer = cosmosClient.GetContainer(cosmosDbMetadata.DatabaseName, string.IsNullOrEmpty(cosmosDbMetadata.LeaseContainerName) ? CosmosDBTriggerConstants.DefaultLeaseCollectionName : cosmosDbMetadata.LeaseContainerName);
```
Comment (ealsur) [inline src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDbScalerProvider.cs]:
```suggestion
            var leaseContainer = cosmosClient.GetContainer(string.IsNullOrEmpty(cosmosDbMetadata.LeaseDatabaseName) ?cosmosDbMetadata.DatabaseName : cosmosDbMetadata.LeaseDatabaseName, string.IsNullOrEmpty(cosmosDbMetadata.LeaseContainerName) ? CosmosDBTriggerConstants.DefaultLeaseCollectionName : cosmosDbMetadata.LeaseContainerName);
```

File: test/WebJobs.Extensions.CosmosDB.Tests/WebJobs.Extensions.CosmosDB.Tests.csproj (hunk 3 above)
Comment (alrod) [reply test/WebJobs.Extensions.CosmosDB.Tests/WebJobs.Extensions.CosmosDB.Tests.csproj]: I just picked up the higher major version, anyway is a test project.

Comment (ealsur) [reply test/WebJobs.Extensions.CosmosDB.Tests/WebJobs.Extensions.CosmosDB.Tests.csproj]: Yes but 7 is not the LTS. This is a net6 app, not net7, shouldn't we ideally use 6.X?

Comment (alrod) [reply test/WebJobs.Extensions.CosmosDB.Tests/WebJobs.Extensions.CosmosDB.Tests.csproj]: ok, changed to 6.0.0

**Review by ealsur — APPROVED** [review]

**Pull request merged**